modules/users/server/controllers/users/users.profile.server.controller.js

```diff
@@ -875,3 +876,130 @@ exports.getUserMemberships = function(req, res) {
       return res.send(memberships);
     });
 };
+
+exports.removePushRegistration = function(req, res) {
+
+  if (!req.user) {
+    return res.status(403).send({
+      message: errorHandler.getErrorMessageByKey('forbidden')
+    });
+  }
+
+  var user = req.user;
+  var token = req.params.token;
+
+  var query = {
+    $pull: {
+      pushRegistration: {
+        token: token
+      }
+    }
+  };
+
+  User.findByIdAndUpdate(user._id, query, {
+    safe: true, // @link http://stackoverflow.com/a/4975054/1984644
+    new: true // get the updated document in return
+  })
+  .exec(function(err, user) {
+    if (err) {
+      return res.status(400).send({
+        message: errorHandler.getErrorMessage(err) || 'Failed to remove registration, please try again.'
+      });
+    } else {
+      return res.send({
+        message: 'Removed registration.',
+        user: user
+      });
+    }
+  });
+
+};
+
+/**
+ * Add push registration
+ */
+exports.addPushRegistration = function(req, res) {
+
+  if (!req.user) {
+    return res.status(403).send({
+      message: errorHandler.getErrorMessageByKey('forbidden')
+    });
+  }
+
+  var user = req.user;
+  var token = req.body.token;
+  var platform = req.body.platform;
+
+  async.waterfall([
+
+    // Remove any existing registrations for this token
+
+    function(done) {
+      User.findByIdAndUpdate(user._id, {
+        $pull: {
+          pushRegistration: {
+            token: token
+          }
+        }
+      }).exec(function(err) {
+        done(err);
+      });
+
+    },
+
+    // Add new registration
+
+    function(done) {
+      User.findByIdAndUpdate(user._id, {
+        $push: {
+          pushRegistration: {
+            platform: platform,
+            token: token,
+            created: Date.now()
+          }
+        }
+      }, {
+        new: true
+      }).exec(function(err, updatedUser) {
+        if (err) return done(err);
+        user = updatedUser;
+        done();
+      });
+    },
+
+    // Notify the user we just added
+
+    function(done) {
+      if (process.env.NODE_ENV === 'production') return done(); // only in dev...
+
+      pushService.sendUserNotification(user, {
+        title: 'Trustroots',
+        body: 'A ' + platform + ' push device was added!',
+        click_action: 'http://localhost:3000/profile/edit/account'
```

Comment:
This might work here better for flexibility with different setups (Docker IP, https):

```js
var protocol = (config.https === true || req.protocol === 'https') ? 'https' : 'http';
var host = protocol + '://' + req.get('host'); // http://localhost:3000
```

...or actually this might be already available as `res.locals.hostPort`